tests/python/test_fields_builder.py

```diff
@@ -6,171 +6,177 @@
+        for i in range(shape):
             x[i] = i * 2
-        for i in range(n):
+        for i in range(shape):
             y[i] = i + 5
-        for i in range(n):
+        for i in range(shape):
             z[i] = i + 10
 
-    func2()
-    for i in range(n):
+    assign_field_multiple()
+    for i in range(shape):
         assert x[i] == i * 2
         assert y[i] == i + 5
         assert z[i] == i + 10
 
-    func1()
-    for i in range(n):
+    assign_field_single()
+    for i in range(shape):
         assert x[i] == i * 3
 
 
-@ti.test(arch=[ti.cpu, ti.cuda, ti.metal])
+@ti.test(arch=[ti.cpu, ti.cuda, ti.vulkan, ti.metal])
 def test_fields_builder_pointer():
-    n = 5
-
+    shape = 5
     fb1 = ti.FieldsBuilder()
     x = ti.field(ti.f32)
-    fb1.pointer(ti.i, n).place(x)
+    fb1.pointer(ti.i, shape).place(x)
     fb1.finalize()
 
     @ti.kernel
-    def func1():
-        for i in range(n):
+    def assign_field_single():
+        for i in range(shape):
             x[i] = i * 3
 
-    func1()
-    for i in range(n):
+    assign_field_single()
+    for i in range(shape):
         assert x[i] == i * 3
 
     fb2 = ti.FieldsBuilder()
     y = ti.field(ti.f32)
-    fb2.pointer(ti.i, n).place(y)
+    fb2.pointer(ti.i, shape).place(y)
     z = ti.field(ti.f32)
-    fb2.pointer(ti.i, n).place(z)
+    fb2.pointer(ti.i, shape).place(z)
     fb2.finalize()
 
-    # test range-for
     @ti.kernel
-    def func2():
-        for i in range(n):
+    def assign_field_multiple0():
+        # test range-for
+        for i in range(shape):
             x[i] = i * 2
-        for i in range(n):
+        for i in range(shape):
             y[i] = i + 5
-        for i in range(n):
+        for i in range(shape):
             z[i] = i + 10
 
-    func2()
-    for i in range(n):
+    assign_field_multiple0()
+    for i in range(shape):
         assert x[i] == i * 2
         assert y[i] == i + 5
         assert z[i] == i + 10
 
-    # test struct-for
     @ti.kernel
-    def func3():
+    def assign_field_multiple1():
+        # test struct-for
         for i in y:
             y[i] += 5
         for i in z:
             z[i] -= 5
 
-    func3()
-    for i in range(n):
+    assign_field_multiple1()
+    for i in range(shape):
         assert y[i] == i + 10
         assert z[i] == i + 5
 
-    func1()
-    for i in range(n):
+    assign_field_single()
+    for i in range(shape):
         assert x[i] == i * 3
 
 
-@ti.test(arch=[ti.cpu, ti.cuda, ti.vulkan])
-def test_fields_builder_destroy():
-    def A(i):
-        n = i * 10**3
+@pytest.mark.parametrize('test_1d_size', [1, 10, 100])
+@pytest.mark.parametrize('field_type', [ti.f32, ti.i32])
+@ti.test(arch=[ti.cpu, ti.cuda, ti.vulkan, ti.metal])
+def test_fields_builder_destroy(test_1d_size, field_type):
+    # note: currently only consider precison that all platform supported,
+    # more detailed here: https://docs.taichi.graphics/lang/articles/basic/type#supported-primitive-types
```

Comment:
```suggestion
    # See https://docs.taichi.graphics/lang/articles/basic/type#supported-primitive-types for more details.
```

BTW could you move these comments before the test?

Reply:
Fixed.